service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -147,15 +150,18 @@ private List<TableMetadata> generateDatasetTables(
     return switch (dataset.storageSystem()) {
       case TERRA_DATA_REPO -> convertDatarepoTablesToCatalogTables(
           datarepoService.getPreviewTables(user, dataset.storageSourceId()).getTables());
-      case TERRA_WORKSPACE, EXTERNAL -> List.of();
+      case TERRA_WORKSPACE -> convertRawlsTablesToCatalogTables(
```

Comment:
I'm wondering if the conversion methods should go into the service layer, so that the datarepo and rawls services return the same shape of data. That might make more sense, because it would mean that the `DatasetService` wouldn't need to know about specific service API shape.

Reply:
I was thinking the same thing, that the conversion code for both TDR and Rawls should be pushed into the lower service layers. But maybe that could wait for a future refactor? This code is (over) due for that kind of work.

Reply:
Yes, I think it's worth making a ticket, but not blocking this work.